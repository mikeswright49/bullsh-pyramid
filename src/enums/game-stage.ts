export enum GameStage {
    Initiation,
    Memorization,
    Flipping,
    Declaration,
    Bullshit,
    Memory,
    Complete,
    Reveal,
}
