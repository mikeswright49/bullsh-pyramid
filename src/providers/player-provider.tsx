import React from 'react';
import { useState, useEffect } from 'react';
import { Player } from 'types/player';
import { PlayerStore } from 'src/stores/player-store';
export const PlayerContext = React.createContext<Player>(null);

export function PlayerProvider(props: React.PropsWithChildren<{ playerId: string }>) {
    const [player, setPlayer] = useState<Player>();

    useEffect(() => {
        if (!props.playerId) {
            return;
        }
        PlayerStore.subscribeToPlayer(props.playerId, (updatedPlayer) => {
            setPlayer(updatedPlayer);
        });

        return () => PlayerStore.unsubscribeToPlayer();
    }, [props.playerId]);

    return <PlayerContext.Provider value={player}>{props.children}</PlayerContext.Provider>;
}
