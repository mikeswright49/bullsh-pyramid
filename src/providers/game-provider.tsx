import React, { useEffect, useState } from 'react';
import { GameState } from 'types/game-state';
import { GameStore } from 'src/stores/game-store';
import { GameStage } from 'src/enums/game-stage';

export const DEFAULT_GAME_STATE: GameState = {
    id: '',
    gameStage: GameStage.Initiation,
    activeRow: 0,
    activeIndex: 0,
    cardCount: 4,
    flipDelay: 5000,
    players: [],
    tiers: [],
};
export const GameContext = React.createContext<GameState>(DEFAULT_GAME_STATE);

export function GameProvider(props: React.PropsWithChildren<{ gameId: string }>) {
    const { gameId } = props;
    const [gameState, setGameState] = useState<GameState>(DEFAULT_GAME_STATE);

    useEffect(() => {
        if (!gameId) {
            return;
        }

        GameStore.subscribeToGame(gameId, (updatedGameState) => {
            setGameState(updatedGameState);
        });

        return () => GameStore.unsubscribeToGame();
    }, [gameId]);

    return <GameContext.Provider value={gameState}>{props.children}</GameContext.Provider>;
}
