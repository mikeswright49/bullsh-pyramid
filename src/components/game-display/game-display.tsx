import React from 'react';
import { GameStage } from 'src/enums/game-stage';
import { useContext } from 'react';
import { GameBoard } from 'src/components/game-board/game-board';
import { GameContext } from 'src/providers/game-provider';

import { GameInitiation } from './stages/game-initiation';
import { GameDeclaration } from './stages/game-declaration';
import { Leaderboard } from 'src/components/leaderboard/leaderboard';
import { GameComplete } from './stages/game-complete';

type DisplayConfiguration = {
    text: string;
    Component?: () => JSX.Element;
    hideLeaderboard?: boolean;
    hideGameboard?: boolean;
};
const DISPLAY_CONFIGURATION: { [key: number]: DisplayConfiguration } = {
    [GameStage.Initiation]: {
        text: '',
        Component: GameInitiation,
        hideGameboard: true,
        hideLeaderboard: true,
    },
    [GameStage.Memorization]: {
        text: 'You have 30s to remember your cards!',
        hideGameboard: false,
        hideLeaderboard: true,
    },
    [GameStage.Flipping]: {
        text: 'The host is flipping a card',
    },
    [GameStage.Declaration]: {
        text: 'Let the world know if you have it',
        Component: GameDeclaration,
        hideGameboard: false,
        hideLeaderboard: true,
    },
    [GameStage.Bullshit]: {
        text: 'People should know who you think was lieing',
    },
    [GameStage.Reveal]: {
        text: 'We all now know the truth',
    },
    [GameStage.Memory]: {
        text: 'Well which cards did you have?',
    },
    [GameStage.Complete]: {
        text: 'Well time for a new game',
        Component: GameComplete,
    },
};

export function GameDisplay() {
    const gameState = useContext(GameContext);
    const { gameStage } = gameState;

    const displayStage = DISPLAY_CONFIGURATION[gameStage];

    return (
        <div className="row">
            <div className="col-3">
                {displayStage.text && <h2>{displayStage.text}</h2>}
                {displayStage.Component && <displayStage.Component />}
                {!displayStage.hideLeaderboard && <Leaderboard />}
            </div>
            <div className="col-9">{!displayStage.hideGameboard && <GameBoard />}</div>
        </div>
    );
}
