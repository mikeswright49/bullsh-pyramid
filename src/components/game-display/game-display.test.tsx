import React from 'react';
import { Player } from '../../../types/player';
import { GameState } from '../../../types/game-state';
import { MOCK_PLAYER } from '../../__tests__/data/mock-player';
import { MOCK_GAME_STATE } from '../../__tests__/data/mock-game-state';
import { GameStage } from '../../enums/game-stage';
import { render, screen } from '@testing-library/react';
import { GameContext } from '../../providers/game-provider';
import { PlayerContext } from '../../providers/player-provider';
import { GameDisplay } from './game-display';

describe('Game Display', () => {
    let playerMock: Player;
    let gameContextMock: GameState;
    beforeEach(() => {
        playerMock = Object.assign({}, MOCK_PLAYER);
        gameContextMock = Object.assign({}, MOCK_GAME_STATE);
    });

    const testCases = Object.keys(GameStage)
        .map((key) => (Number.isInteger(parseInt(key, 10)) ? null : [key]))
        .filter((x) => !!x);
    test.each(testCases)('it should be able to render the game view for %s', (stage) => {
        gameContextMock.gameStage = GameStage[stage];
        render(
            <GameContext.Provider value={gameContextMock}>
                <PlayerContext.Provider value={playerMock}>
                    <GameDisplay />
                </PlayerContext.Provider>
            </GameContext.Provider>
        );
        expect(screen.getByTestId('game-display')).toMatchSnapshot();
    });
});
