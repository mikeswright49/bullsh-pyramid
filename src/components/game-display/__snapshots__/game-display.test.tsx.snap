// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Display it should be able to render the game view for Assign 1`] = `
<div
  class="row"
  data-testid="game-display"
>
  <div
    class="col-3"
  >
    <h2>
      game.display.stage.assign.text
    </h2>
  </div>
  <div
    class="col-9"
  >
    <div
      class="flex-center-center"
      data-testid="at-game-board"
    >
      <div
        data-testid="at-pyramid"
      >
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-1-5"
              style="background-position: -395px -123px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-3-1"
              style="background-position: -79px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-0-10"
              style="background-position: -790px -0px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-0-2"
              style="background-position: -158px -0px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-3-3"
              style="background-position: -237px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-1-7"
              style="background-position: -553px -123px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-1-0"
              style="background-position: -0px -123px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-2-10"
              style="background-position: -790px -246px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-3-10"
              style="background-position: -790px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-3-7"
              style="background-position: -553px -369px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-1-2"
              style="background-position: -158px -123px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-2-1"
              style="background-position: -79px -246px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-3-12"
              style="background-position: -948px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-0-12"
              style="background-position: -948px -0px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-1-12"
              style="background-position: -948px -123px;"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Game Display it should be able to render the game view for Bullshit 1`] = `
<div
  class="row"
  data-testid="game-display"
>
  <div
    class="col-3"
  >
    <h2>
      game.display.stage.bullshit.text
    </h2>
  </div>
  <div
    class="col-9"
  >
    <div
      class="flex-center-center"
      data-testid="at-game-board"
    >
      <div
        data-testid="at-pyramid"
      >
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-1-5"
              style="background-position: -395px -123px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-3-1"
              style="background-position: -79px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-0-10"
              style="background-position: -790px -0px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-0-2"
              style="background-position: -158px -0px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-3-3"
              style="background-position: -237px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-1-7"
              style="background-position: -553px -123px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-1-0"
              style="background-position: -0px -123px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-2-10"
              style="background-position: -790px -246px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-3-10"
              style="background-position: -790px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-3-7"
              style="background-position: -553px -369px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-1-2"
              style="background-position: -158px -123px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-2-1"
              style="background-position: -79px -246px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-3-12"
              style="background-position: -948px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-0-12"
              style="background-position: -948px -0px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-1-12"
              style="background-position: -948px -123px;"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Game Display it should be able to render the game view for Complete 1`] = `
<div
  class="row"
  data-testid="game-display"
>
  <div
    class="col-3"
  >
    <h2>
      game.display.stage.complete.text
    </h2>
    <div
      class="stack-y-2"
    >
      <button
        class="pure-button pure-button-primary"
      >
        Start a new game with same players
      </button>
      <button
        class="pure-button pure-button-secondary"
      >
        Host a new game
      </button>
    </div>
  </div>
  <div
    class="col-9"
  >
    <div
      class="flex-center-center"
      data-testid="at-game-board"
    >
      <div
        data-testid="at-pyramid"
      >
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-1-5"
              style="background-position: -395px -123px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-3-1"
              style="background-position: -79px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-0-10"
              style="background-position: -790px -0px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-0-2"
              style="background-position: -158px -0px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-3-3"
              style="background-position: -237px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-1-7"
              style="background-position: -553px -123px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-1-0"
              style="background-position: -0px -123px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-2-10"
              style="background-position: -790px -246px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-3-10"
              style="background-position: -790px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-3-7"
              style="background-position: -553px -369px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-1-2"
              style="background-position: -158px -123px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-2-1"
              style="background-position: -79px -246px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-3-12"
              style="background-position: -948px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-0-12"
              style="background-position: -948px -0px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-1-12"
              style="background-position: -948px -123px;"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Game Display it should be able to render the game view for Declaration 1`] = `
<div
  class="row"
  data-testid="game-display"
>
  <div
    class="col-3"
  >
    <h2>
      game.display.stage.declaration.text
    </h2>
    <div
      class="stack-y-2"
    />
  </div>
  <div
    class="col-9"
  >
    <div
      class="flex-center-center"
      data-testid="at-game-board"
    >
      <div
        data-testid="at-pyramid"
      >
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-1-5"
              style="background-position: -395px -123px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-3-1"
              style="background-position: -79px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-0-10"
              style="background-position: -790px -0px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-0-2"
              style="background-position: -158px -0px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-3-3"
              style="background-position: -237px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-1-7"
              style="background-position: -553px -123px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-1-0"
              style="background-position: -0px -123px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-2-10"
              style="background-position: -790px -246px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-3-10"
              style="background-position: -790px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-3-7"
              style="background-position: -553px -369px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-1-2"
              style="background-position: -158px -123px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-2-1"
              style="background-position: -79px -246px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-3-12"
              style="background-position: -948px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-0-12"
              style="background-position: -948px -0px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-1-12"
              style="background-position: -948px -123px;"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Game Display it should be able to render the game view for Flipping 1`] = `
<div
  class="row"
  data-testid="game-display"
>
  <div
    class="col-3"
  >
    <h2>
      game.display.stage.flipping.text
    </h2>
  </div>
  <div
    class="col-9"
  >
    <div
      class="flex-center-center"
      data-testid="at-game-board"
    >
      <div
        data-testid="at-pyramid"
      >
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-1-5"
              style="background-position: -395px -123px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-3-1"
              style="background-position: -79px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-0-10"
              style="background-position: -790px -0px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-0-2"
              style="background-position: -158px -0px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-3-3"
              style="background-position: -237px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-1-7"
              style="background-position: -553px -123px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-1-0"
              style="background-position: -0px -123px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-2-10"
              style="background-position: -790px -246px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-3-10"
              style="background-position: -790px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-3-7"
              style="background-position: -553px -369px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-1-2"
              style="background-position: -158px -123px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-2-1"
              style="background-position: -79px -246px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-3-12"
              style="background-position: -948px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-0-12"
              style="background-position: -948px -0px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-1-12"
              style="background-position: -948px -123px;"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Game Display it should be able to render the game view for Initiation 1`] = `
<div
  class="row"
  data-testid="game-display"
>
  <div
    class="col-3"
  >
    
    <div
      class="stack-y-2"
    >
      <h2>
        To join the game go to:
      </h2>
      <ul>
        <li>
          <a
            href="/game/join/mwif"
          >
            Go to the join screen
          </a>
        </li>
        <li>
          Or type in 
          <strong>
            mwif
          </strong>
           at https://bullsh-pyramid.now.sh/game/join
        </li>
      </ul>
      <h3>
        Players Joined
      </h3>
    </div>
  </div>
  <div
    class="col-9"
  />
</div>
`;

exports[`Game Display it should be able to render the game view for Memorization 1`] = `
<div
  class="row"
  data-testid="game-display"
>
  <div
    class="col-3"
  >
    <h2>
      game.display.stage.memorization.text
    </h2>
  </div>
  <div
    class="col-9"
  >
    <div
      class="flex-center-center"
      data-testid="at-game-board"
    >
      <div
        data-testid="at-pyramid"
      >
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-1-5"
              style="background-position: -395px -123px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-3-1"
              style="background-position: -79px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-0-10"
              style="background-position: -790px -0px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-0-2"
              style="background-position: -158px -0px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-3-3"
              style="background-position: -237px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-1-7"
              style="background-position: -553px -123px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-1-0"
              style="background-position: -0px -123px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-2-10"
              style="background-position: -790px -246px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-3-10"
              style="background-position: -790px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-3-7"
              style="background-position: -553px -369px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-1-2"
              style="background-position: -158px -123px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-2-1"
              style="background-position: -79px -246px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-3-12"
              style="background-position: -948px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-0-12"
              style="background-position: -948px -0px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-1-12"
              style="background-position: -948px -123px;"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Game Display it should be able to render the game view for Memory 1`] = `
<div
  class="row"
  data-testid="game-display"
>
  <div
    class="col-3"
  >
    <h2>
      game.display.stage.memory.text
    </h2>
  </div>
  <div
    class="col-9"
  >
    <div
      class="flex-center-center"
      data-testid="at-game-board"
    >
      <div
        data-testid="at-pyramid"
      >
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-1-5"
              style="background-position: -395px -123px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-3-1"
              style="background-position: -79px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-0-10"
              style="background-position: -790px -0px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-0-2"
              style="background-position: -158px -0px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-3-3"
              style="background-position: -237px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-1-7"
              style="background-position: -553px -123px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-1-0"
              style="background-position: -0px -123px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-2-10"
              style="background-position: -790px -246px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-3-10"
              style="background-position: -790px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-3-7"
              style="background-position: -553px -369px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-1-2"
              style="background-position: -158px -123px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-2-1"
              style="background-position: -79px -246px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-3-12"
              style="background-position: -948px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-0-12"
              style="background-position: -948px -0px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-1-12"
              style="background-position: -948px -123px;"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Game Display it should be able to render the game view for Reveal 1`] = `
<div
  class="row"
  data-testid="game-display"
>
  <div
    class="col-3"
  >
    <h2>
      game.display.stage.reveal.text
    </h2>
  </div>
  <div
    class="col-9"
  >
    <div
      class="flex-center-center"
      data-testid="at-game-board"
    >
      <div
        data-testid="at-pyramid"
      >
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-1-5"
              style="background-position: -395px -123px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-3-1"
              style="background-position: -79px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-0-10"
              style="background-position: -790px -0px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-0-2"
              style="background-position: -158px -0px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-3-3"
              style="background-position: -237px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-1-7"
              style="background-position: -553px -123px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-1-0"
              style="background-position: -0px -123px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-2-10"
              style="background-position: -790px -246px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-3-10"
              style="background-position: -790px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card cardHidden"
              data-testid="card-3-7"
              style="background-position: -553px -369px;"
            />
          </span>
        </div>
        <div
          class="flex-center-center stack-y-3"
        >
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-1-2"
              style="background-position: -158px -123px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-2-1"
              style="background-position: -79px -246px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-3-12"
              style="background-position: -948px -369px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-0-12"
              style="background-position: -948px -0px;"
            />
          </span>
          <span
            class="stack-x-4"
          >
            <div
              class="flex-center-center card"
              data-testid="card-1-12"
              style="background-position: -948px -123px;"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
